src/phylum/init/sig.py

```diff
@@ -1,118 +1,73 @@
+        7DXvQb7qwzZT0H11K9Pw8hIHBvZPM+d61GWgWDc3k/rFwMmqd+kytVZy0mVxNdv4
+        P2qvy6BNaiUI7yoB1ahR/6klfkPit0X7pkK9sTHwW+/WcYitTQKnEnRzA3q8EmA7
+        rbU/sFEypzBA3C3qNJZyKSwy47kWXhC4xXUS2NXvew4FoVU6ybMoeDApwsx1AgTu
+        CPPnPlCwuCIyUPezCP5XYczuHfaWeuwArlwdJFSUpMTc+SqO6REKgL9yvpqsO5Ia
+        sQIDAQAB
+        -----END PUBLIC KEY-----
+        """
+    ),
+    encoding="ASCII",
+)
+
+
+def verify_sig(file_path: Path, sig_path: Path) -> None:
+    """Verify a given file has a valid signature.
 
     `file_path` is the path to the file data to verify.
-    `sig_path` is the path to the `.minisig` file containing the minisign signature information.
+    `sig_path` is the path to the `.signature` file containing the RSA SHA256 signature information.
 
-    The public key is an assumed constant, the Minisign Public Key for Phylum, Inc.
+    The public key is an assumed constant, the RSA Public Key for Phylum, Inc.
     """
     try:
-        phylum_public_key = ed25519.Ed25519PublicKey.from_public_bytes(PHYLUM_MINISIGN_PUBKEY_ED25519)
+        # The `cryptography` library does not know this is an RSA public key yet...make sure it is
+        phylum_public_key: types.PUBLIC_KEY_TYPES = serialization.load_pem_public_key(PHYLUM_RSA_PUBKEY)
+        if isinstance(phylum_public_key, rsa.RSAPublicKey):
+            phylum_rsa_public_key: rsa.RSAPublicKey = phylum_public_key
+        else:
+            raise RuntimeError(f"The public key was expected to be RSA but instead got: {type(phylum_public_key)}")
     except UnsupportedAlgorithm as err:
-        raise RuntimeError("Ed25519 algorithm is not supported by the OpenSSL version `cryptography` is using") from err
+        openssl_ver = backend.openssl_version_text()
+        msg = f"The serialized key type is not supported by the OpenSSL version `cryptography` is using: {openssl_ver}"
+        raise RuntimeError(msg) from err
+    except ValueError as err:
+        raise RuntimeError("The PEM data's structure could not be decoded successfully") from err
 
-    signature_algorithm, key_id, signature, trusted_comment, global_signature = extract_minisig_elements(sig_path)
-
-    if signature_algorithm != b"Ed":
-        raise RuntimeError("Only the legacy `Ed` signature algorithm is used by Phylum currently")
-
-    if key_id != PHYLUM_MINISIGN_PUBKEY_KEY_ID:
-        raise RuntimeError("The `key_id` from the `.minisig` signature did not match the `key_id` from the public key")
-
-    # Confirm the trusted comment in the sig_path with the `global_signature` there
+    # Confirm the data from `file_path` with the signature from the `sig_path`
     try:
-        phylum_public_key.verify(global_signature, signature + trusted_comment)
+        print(f" [*] Verifying {file_path} with signature from {sig_path} ...", end="")
+        phylum_rsa_public_key.verify(sig_path.read_bytes(), file_path.read_bytes(), padding.PKCS1v15(), hashes.SHA256())
```

Comment:
Just to be 100% explicit, this function throws an exception when verification fails, correct?

I'm just asking because it feels somewhat sketchy to have a verification like that without an explicit branch on the method's result.

The docs seem to suggest this is the only option though I wonder if we should catch a generic exception maybe? If the exception changed to something else, I'd assume the application would still terminate just without the explicit runtime error?

Reply:
Correct, the [verify method (documentation)](https://cryptography.io/en/latest/hazmat/primitives/asymmetric/rsa/#cryptography.hazmat.primitives.asymmetric.rsa.RSAPublicKey.verify) does not have a return value, but will raise an exception when the signature does not validate. A comment will be added to make this more clear.

If the exception changes, it will still cause the program to terminate. A generic exception catch block can be added.